src/vs/workbench/browser/parts/quickopen/quickOpenController.ts

```diff
@@ -549,6 +568,33 @@ export class QuickOpenController extends WorkbenchComponent implements IQuickOpe
 		return promiseCompletedOnHide;
 	}
 
+	private handleOnCancel(isPicker: boolean): void {
+		// restore the editor part state after cancelling
+		this.historyService.block(true);
```

Comment:
without duplicating code in three places handling the openEditor and blocking of history, services, I need to inject or get access to the historyService in `EditorHistoryEntry`.  All the classes that extend it seem to handle the injection fine with the exception of `GotoLineEntry`.  Thoughts on how to handle that?